src/aggregations/top-hits-aggregation.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Internal counter for added top hits aggregations.
+ * @type {Number}
+ */
+let count = 0;
+
+/**
+ * Construct a Top hits aggregation.
+ *
+ * @param  {Object} opts  Options to include in the aggregation (maybe empty).
+ * @param  {String} name  Aggregation name. Defaults to agg_top_hits_{count}.
+ * @return {Object}       Top hits Aggregation.
+ */
+export default function topHitsAggregation(opts, name) {
+  name = name || `agg_top_hits_${count++}`
```

Comment:
Not sure I'm a fan of maintaining that counter state in this module, this is why in the tests the first aggregation is called `agg_top_hits_0` and the second is `agg_top_hits_1`, even though they're not related to each other. Perhaps the AggregationBuilder should be responsible for renaming aggregations with the same name? Or we can append a random number to all agg names? I can't think of a better solution at the moment.

Instead of making this a blocker maybe we can remove the counter and defer this complexity for now. In the meantime people can always use the custom agg `name` argument to make unique names.